metadata/src/utils/validation.rs

```diff
@@ -262,20 +295,29 @@ fn get_extrinsic_hash(
         bytes = concat_and_hash4(
             &bytes,
             &hash(signed_extension.identifier.as_bytes()),
-            &get_type_hash(registry, signed_extension.extra_ty, &mut cache),
-            &get_type_hash(registry, signed_extension.additional_ty, &mut cache),
+            &get_type_hash_with_precomputed(
+                registry,
+                signed_extension.extra_ty,
+                outer_enums_hashes,
+            ),
+            &get_type_hash_with_precomputed(
+                registry,
+                signed_extension.additional_ty,
+                outer_enums_hashes,
+            ),
         )
     }
 
     bytes
 }
 
 /// Obtain the hash representation of the `frame_metadata::v15::OuterEnums`.
+/// Also returns a HashMap containing the individual hashes of the 3 otuer enums.
 fn get_outer_enums_hash(
     registry: &PortableRegistry,
     enums: &OuterEnumsMetadata,
     only_these_variants: Option<&[&str]>,
-) -> [u8; HASH_LEN] {
+) -> ([u8; HASH_LEN], HashMap<u32, [u8; HASH_LEN]>) {
```

Comment:
A separate type for the the `hash` + individual hashes for the outer enums hashes could maybe make sense.
Because documenting this is a little tricky